bindings/py/cpp_src/bindings/engine/py_utils.hpp

```diff
@@ -41,6 +41,16 @@ namespace nupic_ext {
     template<typename T> T* get_end(py::array_t<T>& a) { return ((T*)a.request().ptr) + a.size(); }
 
     template<typename T> T* get_it(py::array& a) { return (T*)a.request().ptr; }
+
+    // Check that the precision in bytes matches the data size of the array
+    template<typename T> T* get_it( int precision, py::array& a)
+	{
+	if (precision != a.request().itemsize)
```

Comment:
can we use only the signature `get_it<T>(py:array& a)` 
removing the precision field, and instead 
`NTA_CHECK(sizeof(T) == a.request().itemsize) << "Invalid numpy array precision used.";` //change to "Numpy type must match with C++" << string(T);   ?

Reply:
and if there still is `get_it(py:array_t<T>& a);` remove that and only use the correct (precision-checking method)?

Reply:
CC @fcr

Reply:
also, the NTA_CHECK is optional, if the `throw ` works better for handling in python (?)

Reply:
Gentlemen,
Your improvements are well beyond what I would come up with in C++ ( I am fully a untyped or inferred type language person).
I will only use the community version via Python.